### PR TITLE
WiP: Add move_node behaviors

### PR DESCRIPTION
**Pull request by iwburns**
@Drakulix Here's my first draft on re-factoring move_node_[whatever] into the new behavior format.

Do you have any thoughts on this?  I'm probably going to do a bit more on the docs side of things here, but otherwise I'm mostly done.

### CODE REVIEW

**Comment by Drakulix** [comment]
Sorry for the late reply. Looks good to me!

**Comment by iwburns** [comment]
@Drakulix no problem and cool deal!  I'll take a glance at this shortly and merge it if I don't see anything strange (it's been a few days since I've looked at this).

File: src/behaviors.rs [hunk 1]
```diff
@@ -13,10 +13,9 @@ pub enum RemoveBehavior {
...
     ///
     /// ```
-    /// use id_tree::Tree;
-    /// use id_tree::Node;
-    /// use id_tree::RemoveBehavior;
-    ///
+    /// # use id_tree::Tree;
+    /// # use id_tree::Node;
+    /// # use id_tree::RemoveBehavior;
     /// let mut tree: Tree<i32> = Tree::new();
```
Comment (iwburns) [inline src/behaviors.rs]: @Drakulix what do you think about me commenting out these `use` statements?  I thought they were a bit repetitive and just adding more noise to the doc tests.

**Comment by iwburns** [comment]
@Drakulix other than my inline comment here, I'm ready to merge this.  I think the only reason the appveyor build failed is because there's no configuration file for it in this branch yet.

File: src/behaviors.rs (hunk 1 above)
Comment (Drakulix) [inline src/behaviors.rs]: I would like to see this to be consistent with all the other tests. Its weird if you can copy&paste some examples as they are, but others not.
Alternatively you could shorten the import to `use id_Tree::{Tree, Node, RemoveBehavior};` or `use id_tree::*`.

Comment (iwburns) [inline src/behaviors.rs]: Yeah,  that's a good point.  And I think it's probably fine to just use `*` in the examples.  I'll go with that solution for now.  Thanks!

**Comment by iwburns** [comment]
@Drakulix Sorry, I know I keep going back and forth on this one.

The more I read `LocationBehavior` the more I don't really like it that much.  I think we should change it to something like `Location` which is more generic and less "wordy", or we should split it up into a `MoveBehavior` and an `InsertBehavior` if we want it to be more specific to that function.

Since we're already going towards the path of making it more generic and usable in more places, I think something like `Location` is better.  We can still leave it in the `behaviors` module, but I don't really like the `Behavior` suffix for some reason (unless it's prefixed with something specific to the function that it goes with).

Thoughts on this?

**Comment by iwburns** [comment]
Another thing to consider here is that if we keep `move` and `insert` using the same enum for their behaviors, then if we ever want to add a behavior to one of them that doesn't really apply to the other, we won't really be able to do that very easily.

**Comment by Drakulix** [comment]
That's actually a though choice, but you are right about this.
I think I would actually prefer to have this consistent with all other functions and just name it `InsertBehavior` and `MoveBehavior`.

**Comment by iwburns** [comment]
@Drakulix alright, at this point I feel good about this one.  Do you have any thoughts or changes to get in here before I merge it?

**Comment by Drakulix** [comment]
No. Seems alright.

**Pull request merged**